src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntityDueDiligenceValidator.java

```diff
@@ -0,0 +1,101 @@
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.StringValidators;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class OverseasEntityDueDiligenceValidator {
+
+    private final AddressDtoValidator addressDtoValidator;
+
+    @Autowired
+    public OverseasEntityDueDiligenceValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+
+    public Errors validate(OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto, Errors errors, String loggingContext) {
+        validateIdentityDate(overseasEntityDueDiligenceDto.getIdentityDate(), errors, loggingContext);
+        validateName(overseasEntityDueDiligenceDto.getName(), errors, loggingContext);
+        validateAddress(overseasEntityDueDiligenceDto.getAddress(), errors, loggingContext);
+        validateEmail(overseasEntityDueDiligenceDto.getEmail(), errors, loggingContext);
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getSupervisoryName())) {
+            validateSupervisoryName(overseasEntityDueDiligenceDto.getSupervisoryName(), errors, loggingContext);
+        }
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getAmlNumber())) {
+            validateAmlNumber(overseasEntityDueDiligenceDto.getAmlNumber(), errors, loggingContext);
+        }
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getPartnerName())) {
+            validatePartnerName(overseasEntityDueDiligenceDto.getPartnerName(), errors, loggingContext);
+        }
+        return errors;
+    }
+
+    private boolean validateIdentityDate(LocalDate identityDate, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(
+                OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE,
+                OverseasEntityDueDiligenceDto.IDENTITY_DATE_FIELD);
+        return UtilsValidators.isNotNull(identityDate, qualifiedFieldName, errors, loggingContext) &&
+                DateValidators.isDateIsInPast(identityDate, qualifiedFieldName, errors, loggingContext) &&
+                DateValidators.isDateIsWithinLast3Months(identityDate, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateName(String name, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(
+                OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE,
+                OverseasEntityDueDiligenceDto.NAME_FIELD);
+        return StringValidators.isNotBlank(name, qualifiedFieldName, errors, loggingContext) &&
+               StringValidators.isLessThanOrEqualToMaxLength(name, 256, qualifiedFieldName, errors, loggingContext) &&
```

Comment:
The 256 constants should go in a new ValidationConstants class.

Reply:
This might be difficult because of all the different max lengths for different models. It would be better to confine them to the class itself.

Reply:
ok, just create a class level constant in that class so the 256 doesn't get repeated.